DS/DSAssignmrnt8/Sorting.java

```diff
@@ -0,0 +1,121 @@
+public class Sorting {
+
+	private int[] numbers;
+	private int number;
+
+	public int[] bubbleSort(int a[]) {
+		boolean swapped;
+		int indexOfLastUnsortedElement = a.length - 1;
+		do {
+			swapped = false;
+			for (int i = 0; i < indexOfLastUnsortedElement; i++) {
+
+				if (a[i] > a[i + 1]) {
+					int temp;
+					temp = a[i];
+					a[i] = a[i + 1];
+					a[i + 1] = temp;
+					swapped = true;
+
+				}
+
+			}
+
+			indexOfLastUnsortedElement--;
+
+		} while (swapped);
+		return a;
+	}
+
+	public int[] quicksort(int[] a) {
+
+		if (a == null || a.length == 0) {
+			return a;
+		}
+		numbers = a;
+		number = a.length;
+		quicksort(0, number - 1);
+		return numbers;
+	}
+
+	private void quicksort(int low, int high) {
+		int i = low, j = high;
+
+		int pivot = numbers[low];
+
+		while (i <= j) {
+			while (numbers[i] < pivot) {
+				i++;
+			}
+			while (numbers[j] > pivot) {
+				j--;
+			}
+
+			if (i <= j) {
+
+				int temp = numbers[i];
+				numbers[i] = numbers[j];
+				numbers[j] = temp;
+
+				i++;
+				j--;
+			}
+		}
+
+		if (low < j)
+			quicksort(low, j);
+		if (i < high)
+			quicksort(i, high);
+	}
+
+	public int[] radixSort(int a[]) {
+
+		int count, first = a[0], exp = 1, n = a.length;
```

Comment:
As u know that 'count' in this situation will never be > 10, you should use 'byte' instead of 'int'; it saves memory.